lib/ansible/modules/cloud/azure/azure_rm_snapshot.py

```diff
@@ -0,0 +1,378 @@
+        type: str
+  state:
+    description:
+      - Assert the state of the Snapshot.
+      - >-
+        Use C(present) to create or update an Snapshot and C(absent) to delete
+        it.
+    default: present
+    choices:
+      - absent
+      - present
+extends_documentation_fragment:
+  - azure
+  - azure_tags
+author:
+  - Zim Kalinowski (@zikalino)
+
+'''
+
+EXAMPLES = '''
+- name: Create a snapshot by importing an unmanaged blob from the same subscription.
+  azure_rm_snapshot:
+    resource_group: myResourceGroup
+    name: mySnapshot
+    location: eastus
+    creation_data:
+      create_option: Import
+      source_uri: 'https://mystorageaccount.blob.core.windows.net/osimages/osimage.vhd'
+'''
+
+RETURN = '''
+id:
+  description:
+    - Resource Id
+  returned: always
+  type: str
+  sample: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/Microsoft.Compute/snapshots/mySnapshot
+'''
+
+import time
+import json
+import re
+from ansible.module_utils.azure_rm_common_ext import AzureRMModuleBaseExt
+from ansible.module_utils.azure_rm_common_rest import GenericRestClient
+from copy import deepcopy
+try:
+    from msrestazure.azure_exceptions import CloudError
+except ImportError:
+    # this is handled in azure_rm_common
+    pass
+
+
+class Actions:
+    NoAction, Create, Update, Delete = range(4)
+
+
+class AzureRMSnapshots(AzureRMModuleBaseExt):
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                updatable=False,
+                disposition='resourceGroupName',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                updatable=False,
+                disposition='snapshotName',
+                required=True
+            ),
+            location=dict(
+                type='str',
+                updatable=False,
+                disposition='/',
+                required=True
+            ),
+            sku=dict(
+                type='dict',
+                disposition='/',
+                options=dict(
+                    name=dict(
+                        type='str',
+                        choices=['Standard_LRS',
+                                 'Premium_LRS',
+                                 'Standard_ZRS']
+                    )
+                )
+            ),
+            os_type=dict(
+                type='str',
+                disposition='/properties/osType',
+                choices=['Windows',
+                         'Linux']
+            ),
+            creation_data=dict(
+                type='dict',
+                disposition='/properties/creationData',
+                required=True,
+                options=dict(
+                    create_option=dict(
+                        type='str',
+                        disposition='createOption',
+                        choices=['Empty',
```

Comment:
add to doc

Reply:
fixed, and added Import as default and only option